narwhal/primary/src/synchronizer.rs

```diff
@@ -381,14 +461,20 @@ impl Synchronizer {
         certificate: Certificate,
         network: &Network,
         sanitize: bool,
+        early_suspend: bool,
     ) -> DagResult<()> {
         let digest = certificate.digest();
         if self.inner.certificate_store.contains(&digest)? {
             trace!("Certificate {digest:?} has already been processed. Skip processing.");
             self.inner.metrics.duplicate_certificates_processed.inc();
             return Ok(());
         }
-
+        if early_suspend {
```

Comment:
The intention is not to avoid locking, but to perform a parents check for a certificate regardless of current in-memory state, and verify consistency between the result and the in-memory state: if the certificate has no missing parent, it should not be suspended; otherwise the missing parents store in memory should match the result from the check.

There is a comment about this above `try_accept_fetched_certificate()`. Adding pointer to that comment here.

Reply:
Ok got it. To be honest it takes some time to identify the various paths where the certificates might come from and the intention behind the various checks we want to make. Probably on follow up PRs we might need to make those a bit more obvious. I think your idea on having a different type for verified certificates (ex `VerifiedCertificate`) makes sense and we can probably expand on it to better capture the path/state

Reply:
Also, apart from the in memory-store state comparison, if we still wanted to go ahead and allow early suspension during the processing when `try_accept_fetched_certificate` that wouldn't constitute a problem - correct?

Reply:
Yes allowing early suspension in `try_accept_fetched_certificate ()` wouldn't be a problem. We only loose an efficient way to verify `State`.